controllers/grafanadatasource/datasource_controller.go

```diff
@@ -234,7 +237,30 @@ func (r *GrafanaDatasourceReconciler) manageSuccess(datasources []grafanav1alpha
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *GrafanaDatasourceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	cmHandler := func(o client.Object) []reconcile.Request {
+		if o.GetName() != constants.GrafanaDatasourcesConfigMapName {
+			return nil
+		}
+		ns := o.GetNamespace()
+		list := &grafanav1alpha1.GrafanaDataSourceList{}
+		opts := &client.ListOptions{
+			Namespace: ns,
+		}
+		err := r.Client.List(context.Background(), list, opts)
+		if err != nil {
+			return nil
+		}
+		requests := make([]reconcile.Request, len(list.Items))
+		for i, ds := range list.Items {
+			requests[i] = reconcile.Request{NamespacedName: types.NamespacedName{
+				Namespace: ns,
+				Name:      ds.GetName(),
+			}}
+		}
+		return requests
+	}
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&integreatlyorgv1alpha1.GrafanaDataSource{}).
+		Watches(&source.Kind{Type: &v1.ConfigMap{}}, handler.EnqueueRequestsFromMapFunc(cmHandler)).
```

Comment:
Why not use `Owns(&v1.ConfigMap{})` instead? That would trigger the normal reconcile function where we could check if it's the right configmap.

Reply:
@pb82 
Unfortunately, the owner of the configmap is not a DataSource resource, but a Grafana resource.
So we cannot use `Owns(&v1.ConfigMap{})` in this controller.